### ISSUE TITLE
Don't apply bootstrap styles and submit invalid form?

### ISSUE THREAD

**Issue by GetTaxSolutions** [formly-js/ng-formly#572]
I'm trying to add ngx-formly (current project) in aspnetcore-angular2-universal project of Mark Pieszak.

I added the packages and add the proper modules like the description in readme.md. It looks like that start to work but bootstrap doesn't apply the validation styles over the form. Form example when the field is required it have to go in red state but it didn't. My changes are in fork of your repository:
[.](https://github.com/GetTaxSolutions/aspnetcore-angular2-universal/commits/master)

Can you check why the styles are not applied when the form is invalid and required fields are empty? And why when the form is invalid it process submit handler and doesn't stop it. I'm trying to migrate one old project and rewrite the code to use the same logic. In old version submit button didn't continue in case when the form is not completely valid. Can you check in the project last commit [.](https://github.com/GetTaxSolutions/aspnetcore-angular2-universal/commit/dbbde98c1453c096d30de4379349e26fea6e8f18) What is wrong to have no validations style as well. I added the bootstrap module which are described in the documentation.

Thank you

**Comment by aitboudad** [comment]
make sure you have included bootstrap style [bootstrap.min.css](https://github.com/formly-js/ng-formly/blob/master/demo/src/index.html#L8) in your `index.html`
and define `validationMessages` in the FormlyModule

```ts
FormlyModule.forRoot({
  validationMessages: [
    { name: 'required', message: 'this field is required' },
  ]
}),
```

**Comment by aitboudad** [comment]
why you have defined seveval `formly-form` on your form, I think only one is enough and if you want to use bootstrap grid system you just have to pass `className` in field declaration see the example https://stackblitz.com/edit/angular-5ioeet

**Comment by GetTaxSolutions** [comment]
We had added multiple formly-form to control the design more flexible. That was the reason but It hadn't be a problem in old version for angular 1.x. May be is not problem and here. Actually I forked your example and changed one field to be required. In the example the bootstrap error styles aren't applied as well:
[https://stackblitz.com/edit/angular-ye9nqh](url)

If I change FormlyModule.forRoot parameters to
`
FormlyModule.forRoot({
  validationMessages: [
    { name: 'required', message: 'this field is required' },
  ]
}),`

The message ''this field is required' appears to the box but is not in red as well.How can be set up to use bootstrap error styles? Or we have to add our own styles to show the error fields?

Thank you,
GetTaxSolutions Team

**Comment by aitboudad** [comment]
@GetTaxSolutions because you used an old version of bootstrap, you should include the beta version

```ts
https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.2/css/bootstrap.min.css
```

**Comment by GetTaxSolutions** [comment]
Is there version which is compatible with bootstrap 3.3.7?

**Comment by aitboudad** [comment]
I think we can support both of them, I'll provide a PR tomorrow

**Comment by aitboudad** [comment]
@GetTaxSolutions the support for bootstrap `3.3.7` has been added in `2.0.0-beta.7`

**Comment by GetTaxSolutions** [comment]
I tried to see it in example and modify your with 3.3.7 and 2.0.0-beta.7 - https://stackblitz.com/edit/angular-hvklvh
It looks that styles are correct applied but it check validation on text change, not on blur. Can I setup to check when the input control lost the focus?

**Comment by aitboudad** [comment]
it's a regression, I'm on it.

**Comment by aitboudad** [comment]
fixed in `2.0.0-beta.8`

**Comment by GetTaxSolutions** [comment]
Now looks good in example. Thank you
